### DISCUSSION TITLE
Barn Programming Language!

### DISCUSSION THREAD

**Discussion by SolindekDev** [community/community#49878] [Other Feature Feedback, Questions, & Ideas]
Hi everyone 👋

I was working on a brand new programming language for like 6 months with a little break, the name of it is Barn. I've started to write Barn like at the beginning of October because I was really bored and didn't have any goal.

But more about Barn, It's an compiled programming language to C++. For now it is in 0.1v BETA version, which means it don't pointers, arrays and also format arguments for now, but I want to do them in future. Now im working on 0.2v which is improving everything that I've done wrong in the first version like ugly if conditions, new auto type for variable and some more stuff. Syntax is very easy and here's the example of it

```
@import "std.ba"

fun main() {
   println("Hello World")
}
```

[removed by moderator]

Have a nice day/night! 👺

**Comment by tuves** [comment]
Hi @SolindekDev, we’re glad you’re here 😀

Thanks for sharing your (project/website/etc) with the community. Unfortunately, we currently do not allow self-promotion or advertising on the Community on Discussions. We want to make sure there is space for users to ask questions without overwhelming them with other conversations. Thank you for helping us maintain a productive and tidy community for all our members.

**Comment by SolindekDev** [comment]
Sorry I didn't know that, other people posted self-promotion too so i thought it's okay. 

Cheers Bruno
